addons/hw_drivers/controllers/driver.py

```diff
@@ -1,432 +1,328 @@
+        self.data = ""
+        self.gatt_device = False
 
     def get_name(self):
-        lsusb = str(subprocess.check_output('lsusb')).split("\\n")
-        for usbpath in lsusb:  # Should filter on usb devices or inverse loops?
-            device = self.dev
-            if "%04x:%04x" % (device.idVendor, device.idProduct) in usbpath:
-                return usbpath.split("%04x:%04x" % (device.idVendor, device.idProduct))[1]
-        return str(device.idVendor) + ":" + str(device.idProduct)
+        return ''
+
+    def get_identifier(self):
+        return ''
 
     def get_connection(self):
-        return 'direct'
+        return ''
+
+    def get_message(self):
+        return ''
 
     def value(self):
         return self.value
 
+    def supported(self):
+        pass
 
-#----------------------------------------------------------
-# Bluetooth
-#----------------------------------------------------------
-class GattBtManager(gatt.DeviceManager):
-
-    def device_discovered(self, device):
-        # TODO: need some kind of updated_devices mechanism or not?
-        path = "bt_%s" % (device.mac_address,)
-        if path not in drivers:
-            for driverclass in btdrivers:
-                d = driverclass(device = device, manager=self)
-                if d.supported():
-                    drivers[path] = d
-                    d.connect()
-                    send_iot_box_device(False)
-
+    def action(self, action):
+        pass
 
-class BtManager(Thread):
-    gatt_manager = False
+    def disconnect(self):
+        del all_devices[self.get_identifier()]
 
-    def run(self):
-        dm = GattBtManager(adapter_name='hci0')
-        self.gatt_manager = dm
-        dm.start_discovery()
-        dm.run()
 
 #----------------------------------------------------------
-# Bluetooth drivers
+# Device manager
 #----------------------------------------------------------
-btdrivers = []
 
-class BtMetaClass(type):
-    def __new__(cls, clsname, bases, attrs):
-        newclass = super(BtMetaClass, cls).__new__(cls, clsname, bases, attrs)
-        btdrivers.append(newclass)
-        return newclass
+class DeviceManager():
+    sessions = []
 
+    def addRequest(self, requests):
+        session = {
+            'requests': requests,
+            'event': threading.Event(),
+            'queue': [],
+        }
+        self.sessions.append(session)
+        return session
 
-class BtDriver(Driver, metaclass=BtMetaClass):
+    def deviceChanged(self, device):
+        for session in self.sessions:
+            for req in session['requests']:
+                if device.get_identifier() == req.get('device_id'):
+                    session['queue'].append({
+                        'request_id': req.get('request_id'),
+                        'value': device.value,
+                        'data': device.data,
+                    })
+                    session['event'].set()
 
+class IoTDevice(object):
+    pass
```

Comment:
At least make a __init__ with dev and connection_type, eventually kwargs could be used for magic field creation on initialisation.
```
def __init__(dev, connection_type, **kwargs):
    self.dev = dev
    self.connection_type = connection_type
    ...
```